workshop/posting-error-example.md

```diff
@@ -0,0 +1,199 @@
+- What is the error or problematic outcome?
+
+*Optionally:*
+- What things have you tried thus far?
+
+To demonstrate what we mean by these guidelines we will walk through an example:
+
+# A troubleshooting example
+
+In this example, I'm using our exercise notebook, `04a-intro_to_R_exercise.Rmd`.
+
+Below I'm showing a screenshot of what this error would look like when encountering it in [our RStudio Server](../virtual-setup/rstudio-login.md)
+
+<img src = "screenshots/file.path-error.png" width = "800">
+
+## Step 1) Do what you can to try to solve the error yourself (but don't tire yourself out!)
+
+The best way to learn how to solve errors in code is to figure them out yourself.
+So try to explore the error a bit - but we are also here to help you so don't hesitate to ask for help! (If you are feeling exasperated by your error, skip to Step 2).
+
+We recommend looking over our [debugging guide](https://github.com/AlexsLemonade/training-modules/blob/master/intro-to-R-tidyverse/00b-debugging_resources.md) which can explain what some of the most common errors mean.
+The debugging guide also has tips on first steps to take which may help you get to the root of the problem.
+
+**A short list of things to try:**  
+
+- Try to identify which part of your code appears to be the problem through trying smaller parts of the code and seeing if the error still occurs (called chunking).
+In the chunking example below, each highlighted portion is something you could run by itself to test.
+(Using `Command + Enter` in RStudio will run what you have highlighted.)
+
+<img src = "screenshots/chunking_code.png" width = "500">
+
+- Go back to the beginning of your notebook and re-run your code in order and make sure you haven't missed any vital steps! Order matters!
+
+## Step 2) Draft out your question you'll post.
+
+You may want to open up a text editor to write this out before posting (this is a personal preference thing).
+
+For our example error, here's how each of the "four points" (and the optional one) could be included in your posted question; we will go through the details on each point:
+
+<img src = "screenshots/breakdown.png">
+
+### Where is this error occurring; what notebook/chunk?
+
+- In our workshop, notebook references tend to make sense to provide, however, more generally this should be any context around __where__ you've encountered this error.
+
+- If you are *not* working from any CCDL materials, it is *even more important* to provide context around what the notebook you are creating is supposed to be doing.
+
+- Note that line numbers change as you've added code, so line numbers as a reference aren't as helpful.
+
+*For our example:*  
+<img src = "screenshots/where-is-the-error.png" width = "600">
+
+### What is the goal of this code?  
+
+- Providing context around your end goals for this code will help others tailor their advice toward your goal as well as helping the understand what you are working toward.
```

Comment:
```suggestion
- Providing context around your end goals for this code will help others tailor their advice toward your goal as well as help them understand what you are working toward.
```

Is this what you meant here? Seems a little redundant to me (feel free to disagree!)... unless you really meant
```
- Providing context around your end goals for this code will help others tailor their advice toward your goal as well as help you understand what you are working toward.
```

Reply:
It is a redundant sentence! I'll trim it down! Thx